### ISSUE TITLE
Add support for integrating with the Canonical Observability Stack

### ISSUE THREAD

**Issue by gaborbk** [canonical/nginx-ingress-integrator-operator#182]
### Enhancement Proposal

For application deployments that use the Canonical Observability Stack (COS), it would be useful to see nginx-ingress-integrator  logs, metrics, alerts and Grafana dashboards generated from them alongside the application's observability data. 

From my brief reserach, this should be possible through:
- adding support for the COS Loki `LogForwarder` charm interface: https://charmhub.io/loki-k8s/libraries/loki_push_api
- enabling `nginx-prometheus-exporter` and supporting the COS Prometheus `MetricsEndpointProvider` interface: https://github.com/nginxinc/nginx-prometheus-exporter and https://charmhub.io/prometheus-k8s/libraries/prometheus_scrap
- creating Grafana dashboard(s) and exposing them to COS Grafana through the `grafana_dashboard` relation:  https://charmhub.io/grafana-k8s/libraries/grafana_dashboard
- exposing alert rules under the `loki_alert_rules` and `prometheus_alert_rules` folders in the charm, as described in the Prometheus and Loki documentation: https://charmhub.io/cos-lite

**Comment by amandahla** [comment]
@weiiwang01 Can you give your opinion here, please? Thanks.

**Comment by weiiwang01** [comment]
The `nginx-ingress-integrator` charm is a configurator charm that generates Kubernetes resources, such as ingress resources, service resources, rtc. It doesn't run the Nginx server itself; that’s the responsibility of the Nginx ingress controller. So, I think that observability integration on the `nginx-ingress-integrator` itself wouldn't be very useful.

**Comment by gaborbk** [comment]
I see, thank you for your reply @weiiwang01 If I understand right, the only way to interact with the Nginx server through Juju is using the `nginx-ingress-integrator` charm. If this is the case, what would be the right place to add COS support for Nginx on Juju?

**Comment by weiiwang01** [comment]
This would be difficult, and there is no one-size-fits-all solution. The nginx Ingress Controller is usually managed by the deployment of the Kubernetes cluster, which means that it is often not charmed. For example, if juju is bootstrapped in microk8s, the nginx Ingress Controller will be managed by microk8s and deployed as a Kubernetes service, meaning it is not charmed and cannot be integrated into COS.

**Comment by gaborbk** [comment]
That makes sense, thanks. So would the only somewhat easy solution be to switch to the charmed Traefik operator that supports COS integrations?

**Comment by weiiwang01** [comment]
Yes, if you do not have control over the Kubernetes cluster, the Traefik charm would be a better choice for getting COS integration for ingresses. It basically creates its own ingress for your application instead of using the Kubernetes cluster's ingress.

**Comment by gaborbk** [comment]
Thank you, that sounds like a better solution for my use case then. Please feel free to close this issue.
